---
string hardware_identification
string message
bool success